### ISSUE TITLE
Any plan to support scala 3?

### ISSUE THREAD

**Issue by alessandrocandolini** [d2a4u/meteor#243]
The project is currently cross published for scala 2.12 and 2.13. Are there any plans to start supporting scala3? 
Importing meteor in scala3 projects using `cross(CrossVersion.for3Use2_13)` is not an option, because dependencies have different incompatible artefacts for scala2 and scala3

**Comment by d2a4u** [comment]
hi, sorry for the late reply. Unfortunately there isn't a plan for upgrading from scala2 to scala3 yet. I've tried to do it but it seems to require a lot more effort that expected. Please feel free to raise a PR if you have done this before. Cheers.

**Comment by d2a4u** [comment]
@mhzajac [integration tests](https://github.com/d2a4u/meteor/actions/runs/4579526042/jobs/8087395031?pr=301) are failing for scala 3 and that is the only blocker atm. It looks like this [bug](https://github.com/lampepfl/dotty/pull/16882) but I might be wrong. I haven't managed to figure out why. Any suggestions are welcomed. Thanks.

**Comment by mhzajac** [comment]
Ah, that looks right. I no longer see that error using dotty/main, but it's not officially published yet ☹️

**Comment by marko-asplund** [comment]
Any news on Scala 3 support @d2a4u ?
[3.3.1-RC1](https://github.com/lampepfl/dotty/releases/tag/3.3.1-RC1) includes [dotty#16882](https://github.com/lampepfl/dotty/pull/16882), so it should now be easy to validate the hypothesis that the IT failures were caused by that issue.

**Comment by marko-asplund** [comment]
Update: please see https://github.com/d2a4u/meteor/pull/301#issuecomment-1675922253

**Comment by marko-asplund** [comment]
PR https://github.com/d2a4u/meteor/pull/343 updated to Scala v3.3.1-RC7.
Please have a look @d2a4u and others.

**Comment by marko-asplund** [comment]
Scala [v3.3.1](https://github.com/lampepfl/dotty/releases/tag/3.3.1) was released on 2023-09-08.
Any chance of someone looking at this issue again?
Please also check out https://github.com/d2a4u/meteor/pull/343.

**Comment by viswanath7** [comment]
We've waited too long for Scala 3 support. Please help!

**Comment by d2a4u** [comment]
hi, sorry for the long delay. I've not been able to spend as much time as I did before on this project. Version [2.0.0](https://github.com/d2a4u/meteor/releases/tag/v2.0.0) has been released with Scala 3 support. With this release, I've dropped supports for `scanamo` because when I started working on Scala 3 support, `scanamo` hadn't supported it yet so the removal was needed to unblock `meteor`'s work. I will consider re-adding it in the future. Thank @marko-asplund for helping with the Scala 3 support release.
